### PR TITLE
Use GitHub actions

### PR DESCRIPTION
**Pull request by steven-murray**
This updates the pre-commit dependencies and updates to use Github actions.

### CODE REVIEW

**Comment by steven-murray** [comment]
@MuellerSeb I tried to copy across the `cibuildwheel` stuff from Travis, but it's complaining that we're making a pure-python distribution, so it doesn't need `cibuildwheel`. Why were we using that tool in the first place?

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          python -m pip install --upgrade pip
+          pip install build
+          pip install setuptools_scm wheel
+
+      - name: Create setuptools_scm env variable
+        shell: bash
+        run: |
+          wget https://gist.github.com/plaplant/0902f09e59166bac742bbd554f3cd2f9/raw/make_dev_version.sh -O ../make_dev_version.sh
+          version=$(bash ../make_dev_version.sh)
+          echo "SETUPTOOLS_SCM_PRETEND_VERSION=$version" >> $GITHUB_ENV
```
Comment (MuellerSeb) [inline .github/workflows/deploy.yaml:32]: Do we really need that? I guess this could be dropped with #51

File: .github/workflows/test_suite.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - uses: actions/checkout@main
+        with:
+          fetch-depth: 1
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: set PY
+        run: echo "PY=$(python -c 'import hashlib, sys;print(hashlib.sha256(sys.version.encode()+sys.executable.encode()).hexdigest())')" >> $GITHUB_ENV
```
Comment (MuellerSeb) [inline .github/workflows/test_suite.yaml:34]: What does this?

File: hankel/hankel.py [hunk 3]
```diff
@@ -517,19 +489,6 @@ def final_term_amplitude(cls, f, h, k=None, ndim=2):
             The amplitude of the final term in the sum.
         """
         if k is None:
-            return HankelTransform.G(f, h, k)
+            return HankelTransform.final_term_amplitude(f, h, k)
```
Comment (MuellerSeb) [inline hankel/hankel.py:492]: Why is this part of this PR? But I don't have to strong opinions about it.

File: .github/workflows/test_suite.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      ENV_NAME: tests
+      PYTHON: ${{ matrix.python-version }}
+      OS: ${{ matrix.os }}
+    name: Testing
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (MuellerSeb) [inline .github/workflows/test_suite.yaml:23]: We could also add 3.10 now (needs apostrophes: `'3.10'`)

**Review by MuellerSeb — COMMENTED** [review]
This looks good. Some comments, but I guess some points will be addressed in #51.

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (steven-murray) [reply .github/workflows/deploy.yaml:32]: yeah, this'll be dropped there.

File: .github/workflows/test_suite.yaml (hunk 4 above)
Comment (steven-murray) [reply .github/workflows/test_suite.yaml]: Good point. I've added that.

File: hankel/hankel.py (hunk 3 above)
Comment (steven-murray) [reply hankel/hankel.py:492]: Ah. It really shouldn't be here, I guess. I'll revert it.

File: .github/workflows/test_suite.yaml (hunk 2 above)
Comment (steven-murray) [reply .github/workflows/test_suite.yaml]: Apparently nothing useful, I took it out.

**Comment by steven-murray** [comment]
@MuellerSeb I believe I've resolved all your comments -- thanks for the great review!

**Pull request merged**